client/state/reader/reducer.js

```diff
@@ -36,3 +37,6 @@ export default combineReducers( {
 	teams,
 	thumbnails,
 } );
+export default combinedReducer;
+
+registerReducer( [ 'reader' ], combinedReducer );
```

Comment:
One idea how to do the reducer declaration a bit better:
- keep the `reducer.js` module side-effect-free, just exporting just the reducer. That's nice for unit testing it
- create a new file, `state/reducer/index.js`, which does the registration:
```js
import reader from './reducer';
registerReducer( 'reader', reader );
```
Then, all action files that want to trigger the registration would:
```js
import 'state/reader';
```
without the `/reducer` suffix.

As an API, I like this much better.

Reply:
You know what would be even more awesome? If, with some `package.json` magic, we could ensure that any import from `state/reader/*` automatically ensures that the reducer is registered. Think about DLL libraries and their `LoadLibrary` and `FreeLibrary` functions.

Reply:
> You know what would be even more awesome? If, with some `package.json` magic, we could ensure that any import from `state/reader/*` automatically ensures that the reducer is registered. Think about DLL libraries and their `LoadLibrary` and `FreeLibrary` functions.

Interesting! I'll look into whether this is easily doable. Then we'd only have to manually do this for selectors and action creators that touch multiple top-level branches of state.

Reply:
> I'll look into whether this is easily doable.

I'm afraid it won't be feasible at this moment. We'd have to create a top-level module with initialization and reexports:
```js
import './init';
export { default as reducer } from './reducer';
export { ... } from './actions';
export { ... } from './selectors';
```
The most feasible way to make progress here seems to be packaging it all in a `@wordpress/data` store.

Reply:
I've been looking at the webpack side of things and there doesn't seem to be anything there either. We'd need to write and maintain our own webpack plugin for this. And I do think that being explicit in the dependency graph is probably better.

Reply:
Just saying that this new API is beautiful. Over time we can improve/automate the reducer registration, but I think that this is already a huge leap forward!